api/kubernetes/cli/access.go

```diff
@@ -122,3 +116,29 @@ func (kcl *KubeClient) UpdateNamespaceAccessPolicies(accessPolicies map[string]p
 
 	return err
 }
+
+// GetNonAdminNamespaces retrieves namespaces for a non-admin user, excluding the default namespace if restricted.
+func GetNonAdminNamespaces(userID int, endpoint *portainer.Endpoint, clientFactory *ClientFactory) ([]string, error) {
```

Comment:
I'm also not a huge fan of any request that takes a factory as a parameter - somehow it seems that GH is failing at showing me where this function is used but let's discuss and review it.